core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/SigningService.java

```diff
@@ -89,6 +95,30 @@ public String upcheck() {
 
   @JsonRpcMethod("Filecoin.WalletList")
   public List<String> filecoinWalletList() {
-    return fcKeyIdentifiers.list(KeyType.SECP256K1);
+    return fcSigners.availableIdentifiers().parallelStream().collect(Collectors.toList());
+  }
+
+  @JsonRpcMethod("Filecoin.WalletSign")
+  public Bytes filecoinWalletSign(final String filecoinAddress, final String dataToSign) {
+    // convert dataToSign from Base32 to Bytes
+    final Optional<ArtifactSigner> signer = fcSigners.getSigner(filecoinAddress);
+
+    final ArtifactSignature signature;
+    if (signer.isPresent()) {
+      final Bytes bytesToSign = Bytes.fromBase64String(dataToSign);
+      signature = signer.get().sign(bytesToSign);
+    } else {
+      throw new IllegalArgumentException("No such signer");
+    }
+
+    if (signature.getType() == KeyType.SECP256K1) {
+      final SecpArtifactSignature secpSig = (SecpArtifactSignature) signature;
+      return Bytes.concatenate(
```

Comment:
Is it worth moving the this into a common util somewhere? This almost the same code as being used for the eth secp signature without toHexString call.